### ISSUE TITLE
iCal parser error with invalid continuation lines

### ISSUE THREAD

**Issue by jowlo** [tutao/tutanota#3379]
The iCal format is a mess, but as a user i still want to import my calendars that i exported from my previous calendar application. 

The issue is that continuation lines do not start with a whitespace, see the `LOCATION` property below.

```
BEGIN:VCALENDAR
PRODID:-//BUSINESS-CALENDAR//APPGENIX-SOFTWARE//
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:PUBLISH
X-WR-CALNAME:Test
X-WR-TIMEZONE:Europe/Amsterdam
BEGIN:VEVENT
SUMMARY:test
DTSTART;TZID=Europe/Amsterdam:20210814T120000
DTEND;TZID=Europe/Amsterdam:20210814T130000
DTSTAMP:20210813T050843Z
UID:1234uid@uid.com
SEQUENCE:0
CREATED:20210813T050843Z
DESCRIPTION:
LAST-MODIFIED:20210813T050843Z
LOCATION: This is some
location that is not
correctly indented
STATUS:CONFIRMED
TRANSP:OPAQUE
END:VEVENT
END:VCALENDAR
```

I propose to join a line to the previous one if they do not contain a colon, because we will not be able to parse them anyway because we are expecting `PROP:VALUE` format.

I think it would be better to have slightly ugly imports than to not have imports at all in this case.

**Comment by bedhub** [comment]
What if a such a wrong folded line contains a colon in its content? Maybe we need something like "is not a property line".

I think it might also be fine to just ignore those lines instead so the `LOCATION` would be just "This is some".

**Comment by jowlo** [comment]
> What if a such a wrong folded line contains a colon in its content? Maybe we need something like "is not a property line".

Yes, but i think that is pretty hard to define. That's why the standard has defined lines with starting spaces as continuation lines.

My rationale here was just that when we try to parse every line (after joining together those that start with a whitespace) we throw an error if there is no colon. So we cannot deal with this at all and to be a bit more lenient, we could just add that info to whatever field comes before that.

It seems that Outlook is handling it the same way. I am a bit divided between following the standard and making it easier for users to import their calendars...
